controller/src/main/java/io/pravega/controller/store/stream/tables/TableHelper.java

```diff
@@ -645,4 +645,39 @@ public static boolean isRerunOf(final List<Integer> segmentsToSeal,
 
         return historyRecordOpt;
     }
+
+    public static boolean isScaleInputValid(final List<Integer> segmentsToSeal,
+                                            final List<AbstractMap.SimpleEntry<Double, Double>> newRanges,
+                                            final byte[] segmentTable) {
+
+        List<AbstractMap.SimpleEntry<Double, Double>> oldRanges = segmentsToSeal.stream()
+                .map(segment -> SegmentRecord.readRecord(segmentTable, segment)
+                        .map(x -> new AbstractMap.SimpleEntry<>(x.getRoutingKeyStart(), x.getRoutingKeyEnd())))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toList());
+
+        return reduce(oldRanges).equals(reduce(newRanges));
+    }
+
+    private static List<AbstractMap.SimpleEntry<Double, Double>> reduce(List<AbstractMap.SimpleEntry<Double, Double>> ranges) {
+        List<AbstractMap.SimpleEntry<Double, Double>> result = new ArrayList<>();
+        double low = -1.0;
+        double high = -1.0;
+        for (AbstractMap.SimpleEntry<Double, Double> range : ranges) {
+            if (high < range.getKey()) {
+                if (low != -1.0 && high != -1.0) {
+                    result.add(new AbstractMap.SimpleEntry<>(low, high));
+                }
+                low = range.getKey();
+                high = range.getValue();
+            } else {
+                high = range.getValue();
```

Comment:
I'm not sure this is right, but I could be missing something here. Let's go with one of your examples: `{[0.2, 0.4) [0.3, 0.6)}`. The execution as I read it will be like this:

1- The first range we get is `[0.2, 0.4)`. We have that `high < 0.2`, so we don't add a pair to `result`, but we assign low and high as follows: `low = 0.2` and `high = 0.4`.
2- In the second iteration, we have `[0.3, 0.6)`. We have that `high > 0.3`, so we simply make `high = 0.6`.
3- In line 689, we add the pair `[0.2, 0.6)`, which is the only pair in `result`.

This is different from `{[0.2, 0.4) [0.3, 0.6)}` that you posted above.